as/plugins/org.jboss.ide.eclipse.as.wtp.core/src/org/jboss/ide/eclipse/as/core/server/UnitedServerListenerManager.java

```diff
@@ -71,32 +80,42 @@ public static boolean isJBossServer(IServer server) {
 	}
 
 
-	public synchronized void addListener(UnitedServerListener listener) {
-		if( !list.contains(listener)) {
-			list.add(listener);
-			// united listener manager is already initialized, so init this new listener individually
-			if( isInitialized )
-				initializeListener(listener);
+	public void addListener(UnitedServerListener listener) {
+		boolean requiresInit = false;

```

Comment:
Setting the default value for requiresInit can not be placed inside the synchronize block, because then it is not accessible after the synchronize block is over.  So setting the default is not synchronized, and setting the actual value (4 lines later) is synchronized to ensure it gets the actual value.